### ISSUE TITLE
Gmail Oauth 2.0 authentication

### ISSUE THREAD

**Issue by cianopman** [chatwoot/chatwoot#9158]
### Is your feature or enhancement related to a problem? Please describe.

In fall 2024 Google will remove Less secure apps access in Google Workspace Business including IMAP access with user and password. For a self hosted chatwoot app have you planned to integrate Oauth 2.0 ? What are the possible alternatives? I mean local postfix server o somethings else that I missed in docs?
Thank You

### Describe the solution you'd like

Integrate Google Oauth 2.0 authentication.

**Comment by shivamkb17** [comment]
Setup 2FA in your google account and create app password. 
And use that app password.

**Comment by sanjayws** [comment]
> . For a self hosted chatwoot app have you planned to integrate Oauth 2.0 ? What are the possible alternatives? I mean local postfix server o somethings else that I missed in doc

I believe the request is not for logging in, but to oauth2.0 the email inbox for GMail .

**Comment by cianopman** [comment]
Yes is for oauth2.0 the email inbox for GMail because in fall 2024 Google will remove Less secure apps access in Google Workspace Business including IMAP.

**Comment by pranavrajs** [comment]
@cianopman Yes, this is in the pipeline. We are aware of the deadline, you will see the update soon.

**Comment by mick-net** [comment]
I'm also looking for Google oAuth 2.0 for EMAIL since connecting to IMAP seems not to work. Nont sure if it's because Google stopped accepting new (not-secure) apps (even tough the deadline for existing apps is in the fall), or because it's another error:

```
chatwoot-1  | /gems/ruby/3.2.0/gems/mail-2.8.1/lib/mail/network/retriever_methods/imap.rb:167: warning: DEPRECATED: Call Net::IMAP.new with keyword options
```

full docker console log:
```
chatwoot-1  | I, [2024-07-02T12:32:42.544914 #1]  INFO -- : source=rack-timeout id=7c3ba895-5cfc-44a2-b331-39ea8175b97b timeout=15000ms state=ready
chatwoot-1  | I, [2024-07-02T12:32:42.545594 #1]  INFO -- : [7c3ba895-5cfc-44a2-b331-39ea8175b97b] Started PATCH "/api/v1/accounts/1/inboxes/2" for 192.168.65.1 at 2024-07-02 12:32:42 +0000
chatwoot-1  | I, [2024-07-02T12:32:42.553846 #1]  INFO -- : [7c3ba895-5cfc-44a2-b331-39ea8175b97b] Processing by Api::V1::Accounts::InboxesController#update as JSON
chatwoot-1  | I, [2024-07-02T12:32:42.553966 #1]  INFO -- : [7c3ba895-5cfc-44a2-b331-39ea8175b97b]   Parameters: {"formData"=>false, "channel"=>{"imap_enabled"=>true, "imap_address"=>"imap.google.com", "imap_port"=>992, "imap_login"=>"mick@veraero.nl", "imap_password"=>"[FILTERED]", "imap_enable_ssl"=>true}, "account_id"=>"1", "id"=>"2", "inbox"=>{}}
sidekiq-1   | {"ts":"2024-07-02T12:32:42.582Z","pid":1,"tid":"5i35","lvl":"INFO","msg":"start","ctx":{"class":"ActionCableBroadcastJob","jid":"5a17bdb83404cfb2edcd8881"}}
chatwoot-1  | I, [2024-07-02T12:32:42.582369 #1]  INFO -- : [7c3ba895-5cfc-44a2-b331-39ea8175b97b] [ActiveJob] Enqueued ActionCableBroadcastJob (Job ID: d4852795-9284-459a-8fe0-4df39cd4d1f0) to Sidekiq(critical) with arguments: ["MWFaP6zCVTeVJsbSd9i7vKT5"], "account.cache_invalidated", {:cache_keys=>{:label=>"0000000000", :inbox=>"1719923562", :team=>"0000000000"}, :account_id=>1, :performer=>{:id=>1, :name=>"Mick Vermaat", :available_name=>"Mick Vermaat", :avatar_url=>"", :type=>"user", :availability_status=>"online", :thumbnail=>""}}
sidekiq-1   | I, [2024-07-02T12:32:42.583232 #1]  INFO -- : [ActiveJob] [ActionCableBroadcastJob] [d4852795-9284-459a-8fe0-4df39cd4d1f0] Performing ActionCableBroadcastJob (Job ID: d4852795-9284-459a-8fe0-4df39cd4d1f0) from Sidekiq(critical) enqueued at 2024-07-02T12:32:42Z with arguments: ["MWFaP6zCVTeVJsbSd9i7vKT5"], "account.cache_invalidated", {:cache_keys=>{:label=>"0000000000", :inbox=>"1719923562", :team=>"0000000000"}, :account_id=>1, :performer=>{:id=>1, :name=>"Mick Vermaat", :available_name=>"Mick Vermaat", :avatar_url=>"", :type=>"user", :availability_status=>"online", :thumbnail=>""}}
chatwoot-1  | I, [2024-07-02T12:32:42.583226 #1]  INFO -- : [7c3ba895-5cfc-44a2-b331-39ea8175b97b] [ActiveJob] Enqueued EventDispatcherJob (Job ID: 229bfbc7-6f40-4cd8-b1ea-fbede23fc1f3) to Sidekiq(critical) with arguments: "account.cache_invalidated", Tue, 02 Jul 2024 12:32:42.577396888 UTC +00:00, {:cache_keys=>{:label=>"0000000000", :inbox=>"1719923562", :team=>"0000000000"}, :account=>#<GlobalID:0x0000ffff69713748 @uri=#<URI::GID gid://chatwoot/Account/1>>}
sidekiq-1   | {"ts":"2024-07-02T12:32:42.583Z","pid":1,"tid":"5i3p","lvl":"INFO","msg":"start","ctx":{"class":"EventDispatcherJob","jid":"87065548a6b379e78afd8313"}}
chatwoot-1  | /gems/ruby/3.2.0/gems/mail-2.8.1/lib/mail/network/retriever_methods/imap.rb:167: warning: DEPRECATED: Call Net::IMAP.new with keyword options
sidekiq-1   | I, [2024-07-02T12:32:42.584895 #1]  INFO -- : [ActiveJob] [ActionCableBroadcastJob] [d4852795-9284-459a-8fe0-4df39cd4d1f0] Performed ActionCableBroadcastJob (Job ID: d4852795-9284-459a-8fe0-4df39cd4d1f0) from Sidekiq(critical) in 1.8ms
sidekiq-1   | {"ts":"2024-07-02T12:32:42.585Z","pid":1,"tid":"5i35","lvl":"INFO","msg":"done","ctx":{"class":"ActionCableBroadcastJob","jid":"5a17bdb83404cfb2edcd8881","elapsed":0.003}}
sidekiq-1   | I, [2024-07-02T12:32:42.586424 #1]  INFO -- : [ActiveJob] [EventDispatcherJob] [229bfbc7-6f40-4cd8-b1ea-fbede23fc1f3] Performing EventDispatcherJob (Job ID: 229bfbc7-6f40-4cd8-b1ea-fbede23fc1f3) from Sidekiq(critical) enqueued at 2024-07-02T12:32:42Z with arguments: "account.cache_invalidated", Tue, 02 Jul 2024 12:32:42.577396888 UTC +00:00, {:cache_keys=>{:label=>"0000000000", :inbox=>"1719923562", :team=>"0000000000"}, :account=>#<GlobalID:0x0000ffff8eab3ea0 @uri=#<URI::GID gid://chatwoot/Account/1>>}
sidekiq-1   | I, [2024-07-02T12:32:42.586576 #1]  INFO -- : [ActiveJob] [EventDispatcherJob] [229bfbc7-6f40-4cd8-b1ea-fbede23fc1f3] Performed EventDispatcherJob (Job ID: 229bfbc7-6f40-4cd8-b1ea-fbede23fc1f3) from Sidekiq(critical) in 2.27ms
sidekiq-1   | {"ts":"2024-07-02T12:32:42.586Z","pid":1,"tid":"5i3p","lvl":"INFO","msg":"done","ctx":{"class":"EventDispatcherJob","jid":"87065548a6b379e78afd8313","elapsed":0.003}}
chatwoot-1  | I, [2024-07-02T12:32:42.599277 #1]  INFO -- : source=rack-timeout id=2c2ddacf-a80c-4b84-bc67-e4ea8b21799a timeout=15000ms state=ready
chatwoot-1  | I, [2024-07-02T12:32:42.599660 #1]  INFO -- : [2c2ddacf-a80c-4b84-bc67-e4ea8b21799a] Started GET "/api/v1/accounts/1/inboxes" for 192.168.65.1 at 2024-07-02 12:32:42 +0000
chatwoot-1  | I, [2024-07-02T12:32:42.602216 #1]  INFO -- : [2c2ddacf-a80c-4b84-bc67-e4ea8b21799a] Processing by Api::V1::Accounts::InboxesController#index as JSON
chatwoot-1  | I, [2024-07-02T12:32:42.602265 #1]  INFO -- : [2c2ddacf-a80c-4b84-bc67-e4ea8b21799a]   Parameters: {"account_id"=>"1"}
chatwoot-1  | E, [2024-07-02T12:32:42.606108 #1] ERROR -- : [7c3ba895-5cfc-44a2-b331-39ea8175b97b] [Api::V1::InboxesHelper] check_imap_connection failed with getaddrinfo: Name does not resolve
chatwoot-1  | I, [2024-07-02T12:32:42.620308 #1]  INFO -- : [7c3ba895-5cfc-44a2-b331-39ea8175b97b] Completed 422 Unprocessable Entity in 66ms (Views: 0.2ms | ActiveRecord: 13.4ms | Allocations: 15862)
chatwoot-1  | I, [2024-07-02T12:32:42.621549 #1]  INFO -- : source=rack-timeout id=7c3ba895-5cfc-44a2-b331-39ea8175b97b timeout=15000ms service=77ms state=completed
chatwoot-1  | I, [2024-07-02T12:32:42.625446 #1]  INFO -- : [2c2ddacf-a80c-4b84-bc67-e4ea8b21799a]   Rendered api/v1/accounts/inboxes/index.json.jbuilder (Duration: 17.5ms | Allocations: 8729)
chatwoot-1  | I, [2024-07-02T12:32:42.626401 #1]  INFO -- : [2c2ddacf-a80c-4b84-bc67-e4ea8b21799a] Completed 200 OK in 24ms (Views: 10.6ms | ActiveRecord: 9.4ms | Allocations: 11472)
chatwoot-1  | I, [2024-07-02T12:32:42.626815 #1]  INFO -- : source=rack-timeout id=2c2ddacf-a80c-4b84-bc67-e4ea8b21799a timeout=15000ms service=28ms state=completed

````

**Comment by vishnu-narayanan** [comment]
This is now available starting with `v3.10`. If you are self-hosting, please follow this guide to configure the OAuth app for Gmail.

Ref: https://www.chatwoot.com/docs/self-hosted/configuration/features/email-channel/google-workspace-setup
